### PR TITLE
Refactor the main menu code (Part 1)

### PR DESCRIPTION
**Pull request by sethsimon**
Currently, `game::opening_screen` is hard to read and the logic is hard to follow. It's over 700 lines of code, and the portions relevant to a given task (e.g. making a new character) are spread out in different parts of the function. For example, the function itself is lines 237 - 990, and the portions for loading a game are at lines 501 - 553 and 748 - 822.

My goals are:
- Give each of the non-trivial tabs their own function
- For now, I've added `friend class main_menu;` to `game.h`, but I plan to decouple `game` and `main_menu`
- Cleanup / Readability fixes

### CODE REVIEW

**Comment by illi-kun** [comment]
That's a very good change but please try to keep your PRs as small as possible: it's much easier to review N small PRs than one huge PR.

**Comment by sethsimon** [comment]
Agreed, I went a little overboard with this.

**Comment by sethsimon** [comment]
This is ready.

File: src/main_menu.cpp [hunk 1]
```diff
@@ -363,7 +355,7 @@ bool game::opening_screen()
     mmenu_refresh_motd();
     mmenu_refresh_credits();
 
-    u = player();
+    g->u = player();
```
Comment (codemime) [inline src/main_menu.cpp]: Here and below: `main_menu` must not create the `player` object for the `game` class. `main_menu` should call public methods of the `game`, which then decides what to do with its badly non-encapsulated variable.
